fl4health/strategies/feddg_ga_strategy.py

```diff
@@ -0,0 +1,382 @@
+
+        self.weight_step_size = weight_step_size
+
+        self.train_metrics: Dict[str, Dict[str, Scalar]] = {}
+        self.evaluation_metrics: Dict[str, Dict[str, Scalar]] = {}
+        self.num_rounds: Optional[int] = None
+        self.adjustment_weights: Dict[str, float] = {}
+
+    def configure_fit(
+        self,
+        server_round: int,
+        parameters: Parameters,
+        client_manager: ClientManager,
+    ) -> List[Tuple[ClientProxy, FitIns]]:
+        """
+        Configure the next round of training.
+
+        Will also collect the number of rounds the training will run for in order
+        to calculate the adjustment weight step size. Fails if n_server_rounds is not set in the config
+        or if it's not an integer.
+
+        Args:
+            server_round: (int) the current server round.
+            parameters: (Parameters) the model parameters.
+            client_manager: (ClientManager) The client manager which holds all currently
+                connected clients. It must be an instance of FixedSamplingClientManager.
+
+        Returns:
+            (List[Tuple[ClientProxy, FitIns]]) the input for the clients' fit function.
+        """
+        client_fit_ins = super().configure_fit(server_round, parameters, client_manager)
+
+        assert isinstance(
+            client_manager, FixedSamplingClientManager
+        ), f"Client manager is not of type FixedSamplingClientManager: {type(client_manager)}"
+
+        # Setting self.num_rounds
+        if self.num_rounds is None:
+            assert self.on_fit_config_fn is not None, "on_fit_config_in must be specified"
+            config = self.on_fit_config_fn(server_round)
+            assert "n_server_rounds" in config, "n_server_rounds must be specified"
+            assert isinstance(config["n_server_rounds"], int), "n_server_rounds is not an integer"
+            self.num_rounds = config["n_server_rounds"]
+
+        return client_fit_ins
+
+    def aggregate_fit(
+        self,
+        server_round: int,
+        results: List[Tuple[ClientProxy, FitRes]],
+        failures: List[Union[Tuple[ClientProxy, FitRes], BaseException]],
+    ) -> Tuple[Optional[Parameters], Dict[str, Scalar]]:
+        """
+        Aggregate fit results by weighing them against the adjustment weights and then summing them.
+
+        Collects the fit metrics that will be used to change the adjustment weights for the next round.
+
+        Args:
+            server_round: (int) the current server round.
+            results: (List[Tuple[ClientProxy, FitRes]]) The clients' fit results.
+            failures: (List[Union[Tuple[ClientProxy, FitRes], BaseException]]) the clients' fit failures.
+
+        Returns:
+            (Tuple[Optional[Parameters], Dict[str, Scalar]]) A tuple containing the aggregated parameters
+                and the aggregated fit metrics.
+        """
+
+        _, metrics_aggregated = super().aggregate_fit(server_round, results, failures)
```

Comment:
A comment as to why we're "discarding" the parameters here would be useful, I think.